testcafe/tests/drive/file_sharing_scenario.js

```diff
@@ -0,0 +1,131 @@
+import { Selector, Role } from 'testcafe'
+import { driveUser } from '../helpers/roles'
+import {
+  TESTCAFE_DRIVE_URL,
+  isExistingAndVisibile,
+  getCurrentDateTime,
+  FOLDER_DATE_TIME
+} from '../helpers/utils'
+import Data from '../helpers/data'
+const data = new Data()
+
+import DrivePage from '../pages/drive-model'
+const drivePage = new DrivePage()
+
+import PublicDrivePage from '../pages/drive-model-public'
+const publicDrivePage = new PublicDrivePage()
+
+const file = '[cozy]QA_table_ APPS.xlsx'
+
+//************************
+//Tests when authentified
+//************************
+fixture`File link Sharing Scenario`.page`${TESTCAFE_DRIVE_URL}/`.beforeEach(
+  async t => {
+    await t.useRole(driveUser)
+    await drivePage.waitForLoading()
+  }
+)
+
+test('Drive : Create a $test_date_time folder in Drive', async t => {
+  await drivePage.addNewFolder(FOLDER_DATE_TIME)
+  //We need to pass FOLDER_DATE_TIME through multiple fixture, so we cannot use ctx here.
+})
+
+test('Drive : from Drive, go in a folder, upload a file, and share the file', async t => {
+  await drivePage.goToFolder(FOLDER_DATE_TIME)
+  await drivePage.openActionMenu()
+  await t.pressKey('esc') //close action Menu
+
+  await drivePage.uploadFiles([`../../data/${file}`])
+  await drivePage.shareFirstFilePublicLink()
+
+  const link = await drivePage.copyBtnShareByLink.getAttribute('data-test-url')
+  if (link) {
+    data.sharingLink = link
+    console.log(`SHARING_LINK : ` + data.sharingLink)
+  }
+})
+
+//************************
+// Public (no authentification)
+//************************
+fixture.skip`Drive : Access a folder public link`.page`${TESTCAFE_DRIVE_URL}/`
+  .beforeEach(async t => {
+    await t.useRole(Role.anonymous())
+  })
+  .afterEach(async t => {
+    await publicDrivePage.checkLocalFile('files.zip')
+    await publicDrivePage.deleteLocalFile('files.zip')
+  })
+test('Drive : Access a file public link (desktop)', async t => {
+  await t.navigateTo(data.sharingLink)
+  await publicDrivePage.waitForViewer()
+
+  await publicDrivePage.checkActionMenuPublicDesktop()
+  await t
+    .setNativeDialogHandler(() => true)
+    .click(publicDrivePage.btnPublicDownload)
+    .click(publicDrivePage.btnPublicCreateCozy)
+  await publicDrivePage.checkCreateCozy()
+})
+
+test('Drive : Access a file public link (mobile)', async t => {
+  await t.resizeWindowToFitDevice('iPhone 6', {
+    portraitOrientation: true
+  })
+  await t.navigateTo(data.sharingLink)
+  await publicDrivePage.waitForViewer()
+
+  await publicDrivePage.checkActionMenuPublicMobile()
+  await t
+    .setNativeDialogHandler(() => true)
+    .click(publicDrivePage.btnPublicMobileDownload)
+    .click(publicDrivePage.btnPublicMoreMenu) //need to re-open the more menu
+    .click(publicDrivePage.btnPublicMobileCreateCozy)
```

Comment:
So we have a diff between mobile & not mobile ?

Reply:
Yes, in mobile view, the download button and create cozy button are in a [...] menu